crates/sui-indexer-alt-jsonrpc/src/api/transactions/filter.rs

```diff
@@ -190,33 +175,29 @@ async fn by_checkpoint(
         results.reverse();
     }
 
-    Ok(results)
+    from_digests(*limit, results)
 }
 
 /// Fetch a page of transaction digests that called the described function(s). Functions can be
 /// selected by just their package, their module, or their fully-qualified name. It is an error to
 /// supply a package and function, but no module.
-///
-/// Fetches one more result than was requested to detect a next page.
 async fn tx_calls(
     ctx: &Context,
     page: &Page<u64>,
     package: &ObjectID,
     module: Option<&String>,
     function: Option<&String>,
-) -> Result<Keys, RpcError<Error>> {
+) -> Result<Digests, RpcError<Error>> {
     use tx_calls::dsl as c;
-    use tx_digests::dsl as d;
 
     if let (None, Some(function)) = (module, function) {
         return Err(invalid_params(Error::MissingModule {
             function: function.clone(),
         }));
     }
 
-    let mut query = d::tx_digests
-        .inner_join(c::tx_calls.on(d::tx_sequence_number.eq(c::tx_sequence_number)))
-        .select((d::tx_sequence_number, d::tx_digest))
+    let mut query = c::tx_calls
+        .select(c::tx_sequence_number)
```

Comment:
I really like this approach! It's much cleaner and harder to get things wrong this way.